src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1937,10 +1937,17 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
         const selectString = Object.keys(orderBys)
             .map(orderCriteria => {
                 if (orderCriteria.indexOf(".") !== -1) {
-                    const [aliasName, propertyPath] = orderCriteria.split(".");
```

Comment:
Is it possible to treat the `orderBy` similarily to how we treat `WHERE`?  That way we can do something more generalized?

As it stands, I'd want to gate this behind a postgres check

Reply:
Hmmm, that was my original idea actually, but turns out it's not as straightforward here, because of the generated alias.

But IIRC first I tried to use/change a bit `replacePropertyNames()` but it didn't really work or the change was too invasive. If you think it's worth taking another look, I think implementing a separate similar function for this case would be the way to go.

The way it is now, though I indeed only considered real-life case for Postgres, this is relatively generic as it just makes sure that the entity to table column translation is only done on the string up till the space and what's after the space is left unchanged...